### PR TITLE
Dockerfiles enhanced to support ARM64

### LINKED ISSUE #7700: ARM64 Support
## Is your feature request related to a problem? Please describe

I would like to have GitPod Self-Hosted support ARM64 machines, as I was trying to install it on a cloud-hosted VM and it won't run, due to exec format errors.

## Describe the behaviour you'd like

GitPod will run as a kubernetes application on ARM64 machines.

## Describe alternatives you've considered

None, as I have not found anything else that does what GitPod does **and** supports ARM64 machines for self-hosting.

### PR DESCRIPTION
**Pull request by ZCube**
## Description
Dockerfiles enhanced to support ARM64. GitPod can be built on ARM64.

## Related Issue(s)
Fixes #7700

## How to test
Build GitPod on arm64 using docker image ghcr.io/zcube/workspace-images/gitpod-dev:arm64

To build the frontend, you need the arm64 binary of grpc-tool. It can be found at https://github.com/ZCube/grpc-precompiled-binaries.

replaces package.json
from
```
"grpc-tools": "^1.11.2"
```
to
```
"grpc-tools": "https://zcube.github.io/grpc-precompiled-binaries/grpc-tools.tar.gz"
```
The build was tested using github jobs at https://github.com/zcube/gitpod/tree/arm64-2

After modifying the rendered yaml in the installer, it mostly works like this. 

image

node0 : ARM64
nodex86 : x86

MySQL cannot run on ARM64.

## Release Notes
```release-note
NONE
```

## P.S

* I developed this with RedstoneWizard08 to solve #7700. But suddenly RedstoneWizard08 and the issue disappeared together.

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **furisto**, **jankeromnes** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @furisto @jankeromnes` in a comment when ready.

Associated issue: *#7700*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)**
- **[components/workspacekit/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/workspacekit/OWNERS)**
- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by RedstoneWizard08** [comment]
https://github.com/gitpod-io/gitpod/issues/7700#issuecomment-1028040656

**Comment by RedstoneWizard08** [comment]
Hi! I'm back!
Here's the long version of what happened: https://github.com/gitpod-io/gitpod/issues/7700#issuecomment-1028040656
TL;DR: My account got suspended, I didn't do anything wrong, and everything's okay now. I may work on this issue a little more (the issue has re-appeared) but I think @ZCube has done mostly everything! Thanks so much for your help!

Original issue: https://github.com/gitpod-io/gitpod/issues/7700

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7937#issuecomment-1029102264" title="LGTM">geropl</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **furisto** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @furisto` in a comment when ready.

Associated issue: *#7700*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [geropl]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]
- ~~[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)~~ [geropl]
- **[components/workspacekit/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/workspacekit/OWNERS)**
- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ZCube** [comment]
> 안녕! 내가 돌아왔다! 다음은 발생한 일에 대한 자세한 내용입니다. [#7700 (댓글)](https://github.com/gitpod-io/gitpod/issues/7700#issuecomment-1028040656) TL;DR: 내 계정이 일시 중지되었고 아무 잘못도 없었으며 이제 모든 것이 정상입니다. 이 문제에 대해 조금 더 작업할 수 있지만(문제가 다시 나타남) @ZCube 가 대부분의 작업을 수행했다고 생각합니다! 도와주셔서 정말 감사합니다!
> 
> 원호 : #7700



> Hi! I'm back! Here's the long version of what happened: [#7700 (comment)](https://github.com/gitpod-io/gitpod/issues/7700#issuecomment-1028040656) TL;DR: My account got suspended, I didn't do anything wrong, and everything's okay now. I may work on this issue a little more (the issue has re-appeared) but I think @ZCube has done mostly everything! Thanks so much for your help!
> 
> Original issue: #7700


Thanks to you, I was able to get started, and your point was very helpful. In particular, I received a lot of help in building a test environment using Github Actions.

image

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7937#issuecomment-1029102264" title="LGTM">geropl</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **loujaybee**, **princerachit** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @loujaybee @princerachit` in a comment when ready.

Associated issue: *#7700*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [geropl]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]
- **[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)**
- **[components/image-builder-bob/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-bob/OWNERS)**
- **[components/image-builder-mk3/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-mk3/OWNERS)**
- ~~[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)~~ [geropl]
- **[components/workspacekit/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/workspacekit/OWNERS)**
- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ZCube** [comment]
> /lgtm
> 
> I tested the `amd64` build by starting workspaces and it worked as expected. I also checked the pods/deployments, also as expected. 🎉
> 
> I see this as first step to eventually support ARM platforms. IMO it would make sense to also offer `arm64` images, because it has already started taking shares of cloud space (AWS) and will continue to do so.
> 
> Some thoughts around building additional `arm64` images:
> 
> * we should build them as often as possible (on every build, ideally), without slowing down the build 🐔 🥚
> * to make that work we could:
>   
>   * only build arm64 on `main` builds
>   * run arm64 builds in a parallel job

I finally managed to run it. Does it work using MariaDB? The only one that requires amd64 is MySQL.

**Comment by RedstoneWizard08** [comment]
@ZCube we still need to update the helm chart to support arm64. I've already started doing this. https://github.com/gitpod-io/gitpod/tree/main/chart

**Comment by RedstoneWizard08** [comment]
https://gitpod-arm.github.io/gitpod-arm-chart/helm/

**Comment by csweichel** [comment]
@ZCube please squash the commits - I'll then re-run the CI and I reckon we're good to go

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7937#issuecomment-1029102264" title="LGTM">geropl</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jeanp413**, **kylos101** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jeanp413 @kylos101` in a comment when ready.

Associated issue: *#7700*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [geropl]
- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [geropl]
- **[components/docker-up/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/docker-up/OWNERS)**
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]
- **[components/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide/OWNERS)**
- **[components/ide-proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ide-proxy/OWNERS)**
- **[components/image-builder-bob/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-bob/OWNERS)**
- **[components/image-builder-mk3/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-mk3/OWNERS)**
- ~~[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)~~ [geropl]
- **[components/workspacekit/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/workspacekit/OWNERS)**
- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ZCube** [comment]
@csweichel I squash the commit. I probably won't have to fix it any more. Sorry for bothering you with the commit.

**Comment by princerachit** [comment]
Hi, there seems to be a conflict. Can you please resolve it?

**Comment by ZCube** [comment]
> Hi, there seems to be a conflict. Can you please resolve it?
@princerachit In order to resolve this conflict, I have to modify the part related to the gitpod/openvscode-server-linux-build-agent docker image, but I can't do it. I was lost as to where to start.

**Comment by RedstoneWizard08** [comment]
@ZCube fun little problem. this image doesn't support arm64.
https://hub.docker.com/r/gitpod/openvscode-server-linux-build-agent/tags

**Comment by RedstoneWizard08** [comment]
However, we do have this repo:
https://github.com/gitpod-io/vscode-linux-build-agent

**Comment by RedstoneWizard08** [comment]
https://github.com/gitpod-arm/vscode-linux-build-agent

**Comment by RedstoneWizard08** [comment]
> > Hi, there seems to be a conflict. Can you please resolve it?
> 
> @princerachit In order to resolve this conflict, I have to modify the part related to the gitpod/openvscode-server-linux-build-agent docker image, but I can't do it. I was lost as to where to start.

@ZCube I created a `vscode-linux-build-agent` clone and modified it to support arm64.

https://github.com/gitpod-arm/vscode-linux-build-agent/actions/runs/1814282344

**Comment by akosyakov** [comment]
@csweichel how we are going to maintain it? It seems to add some complexity to IDE images

**Comment by csweichel** [comment]
Hi @ZCube - thank you for the effort you've put into this.
After plenty of internal discussion we've come to the conclusion that at this point we'd rather not merge the PR, because:
- we as Gitpod, and our customers today, don't have the use-case to run Gitpod on ARM64
- we would not be able to guarantee that the ARM64 support would not break down the road, making this a very unreliable feature,
- of the complexity, albeit a surprisingly small number of necessary changes.

I am sorry I don't have better news. Also, the statements above reflect our reasoning at this time, and things might change down the road.
